scripts/mvf_rechunk.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+"""Rechunk an existing MVF dataset"""
+
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()  # noqa: 402
+from builtins import object
+
+import sys
+import os
+import re
+import argparse
+import multiprocessing
+import urllib.parse
+
+from katsdptelstate.rdb_writer import RDBWriter
+import numpy as np
+import dask
+import dask.array as da
+
+from katdal.chunkstore import ChunkStoreError
+from katdal.chunkstore_s3 import S3ChunkStore
+from katdal.chunkstore_npy import NpyFileChunkStore
+from katdal.datasources import TelstateDataSource, view_capture_stream
+from katdal.flags import DATA_LOST
+from katdal.applycal import from_block_function    # TODO: get from dask once available there
+
+
+class RechunkSpec(object):
+    def __init__(self, arg):
+        match = re.match(r'^([A-Za-z0-9_]+)/([A-Za-z0-9_]+):(\d+),(\d+)', arg)
+        if not match:
+            raise ValueError('Could not parse {!r}'.format(arg))
+        self.stream = match.group(1)
+        self.array = match.group(2)
+        self.time = int(match.group(3))
+        self.freq = int(match.group(4))
+        if self.time <= 0 or self.freq <= 0:
+            raise ValueError('Chunk sizes must be positive')
+
+
+class Array(object):
+    def __init__(self, stream_name, array_name, store, chunk_info):
+        self.stream_name = stream_name
+        self.array_name = array_name
+        self.chunk_info = chunk_info
+        self.store = store
+        full_name = store.join(chunk_info['prefix'], array_name)
+        shape = chunk_info['shape']
+        chunks = chunk_info['chunks']
+        dtype = chunk_info['dtype']
+        self.data = store.get_dask_array(full_name, chunks, dtype)
+        self.has_data = store.has_array(full_name, chunks, dtype)
+        self.lost_flags = from_block_function(
+            self._make_lost, shape=shape, chunks=chunks, dtype=np.uint8,
+            name='lost-flags-{}-{}'.format(self.stream_name, self.array_name))
+
+    def _make_lost(self, block_info):
+        loc = block_info['array-location']
+        shape = [l[1] - l[0] for l in loc]
+        if self.has_data[block_info['chunk-location']]:
+            return np.zeros(shape, np.uint8)
+        else:
+            return np.full(shape, DATA_LOST, np.uint8)
+
+
+def get_chunk_store(source, telstate, array):
```

Comment:
What's the main difference - simpler interface? separate store per array?

Reply:
Both I guess.

The code duplication is a bit of a smell though. I'm thinking perhaps I should make `_infer_chunk_store` more generic (take `array` argument that defaults to `correlator_data`) and public, and just call it from here.